### PR TITLE
Improve installation docs

### PR DESCRIPTION
**Pull request by keksipurkki**
### Does any other open PR do the same thing?

Checked open PRs for keywords 'docs' and 'documentationä. None found.

Not relevant.

### What issue is this PR fixing?

None. This is a general improvement to the documentation.

### How did you test this PR?

The PR does not contain executable code. 

No executable code. However, I spent the entire day debugging the maps build for iOS and decided to share the lessons learned. 

This PR contains updates for the installation guide:

- Gave the installation documentation a good formatting scrub.
- Revised the troubleshooting section
- Documented clearly that the library ships with two map backends for iOS (was not clear to me).
 
View the proposed version at:
https://github.com/keksipurkki/react-native-maps/blob/feat/update-docs/docs/installation.md

Thanks for your good work!

### CODE REVIEW

**Comment by todorone** [comment]
Amazing job @keksipurkki, installation instructions are not very user-friendly now, especially for those, who are not very familiar with native mobile builds flow...

File: docs/installation.md [hunk 1]
```diff
@@ -266,56 +298,103 @@ const styles = StyleSheet.create({
  },
 });
 
-export default class MyApp extends React.Component {
- render() {
-   const { region } = this.props;
-   console.log(region);
-
-   return (
-     <View style ={styles.container}>
```
Comment (todorone) [inline docs/installation.md]: redundant space after `style`

Comment (keksipurkki) [reply docs/installation.md]: Fixed.

**Comment by rborn** [comment]
@keksipurkki amazing work, merging ❤️🎉

**Pull request merged**